clients/scheduler/src/components/time-slot-select/index.tsx

```diff
@@ -116,7 +116,7 @@ export const TimeSlotSelect = () => {
   useEffect(() => {
     const hasSlots = dayOfTimeSlots.filter(({ providerSlots }) => providerSlots.length > 0).length > 0
 
-    if (hasSlots && (initialized)) {
+    if ((hasSlots || previousInitializedValue) && initialized) {
```

Comment:
The `hasSlots` check prevents this from firing on first initialization before it has fetched time slots.

The `|| previousInitializedValue` allows it to fire on all subsequent date changes.